src/AlgebraicGeometry/Schemes/AffineSchemeOpenSubscheme/Rings/Methods.jl

```diff
@@ -132,6 +138,14 @@ function ==(a::T, b::T) where {T<:AffineSchemeOpenSubschemeRingElem}
   return true
 end
 
+function Base.hash(a::AffineSchemeOpenSubschemeRingElem, h::UInt)
+  h = hash(parent(a), h)
+  for f in restrictions(a)
+    h = hash(f, h)
+  end
```

Comment:
We should probably not do that. This involves choices of representatives.

Reply:
That should be something that `hash(::typeof(f))` should handle, as the `==` method for `AffineSchemeOpenSubschemeRingElem` delegates to `==` for the restrictions in https://github.com/oscar-system/Oscar.jl/blob/a502167dd30a722be879671bfa71784360302f78/src/AlgebraicGeometry/Schemes/AffineSchemeOpenSubscheme/Rings/Methods.jl#L136

Reply:
The restrictions are elements of localizations of quotient rings. They do not have a good normal form which can be used for hashing and even if they did, one would probably not want to trigger the required GB computation only to get the hash. So I still think we should avoid this altogether. The hash function for these ring elements here is probably forced to stay as dummy as possible.